plugins/rikki/heroeslounge/classes/heroes/HeroUpdater.php

```diff
@@ -276,168 +89,77 @@ public static function updateTalentsForHero($hero, $version = null)
                             ->resize(32, 32)
                             ->save($talent_image_path.DS.$tal->image_url, 100);
                     } else {
-                        //Log::error('Failed to get image for talent '.$talent['title']);
-                        $hotslogs_title = substr($talent['name'], strlen($hero->title));
-                        HeroUpdater::fetchHotslogsImage($tal, $hotslogs_title, $talent['title'], end($temp_string_array), preg_replace("/[^A-Za-z0-9]/", '', $talent['title']));
+                        Log::error('Failed to get image for talent '.$talent['name']);
                     }
                     
-                    
-                    
-                    Log::info('New talent added: '.$talent['title']);
-                } elseif (Talent::where('title', $talent['title'])->where('hero_id', $hero->id)->where('replay_title', 'IS NOT', 'NULL')->count() == 0) {
-                    $tal = Talent::where('title', $talent['title'])->where('hero_id', $hero->id)->firstOrFail();
+                    $tal->suspected_replay_title = preg_replace("/[^A-Za-z0-9]/", '', $tal->title);
                     $tal->replay_title = $talent['name'];
                     $tal->save();
+                    Log::info('New talent added: '.$talent['name']);
+                } elseif (Talent::where('title', $talent['name'])->where('hero_id', $heroModel->id)->where('replay_title', 'IS NOT', 'NULL')->count() == 0) {
```

Comment:
Please add a comment explaining what is this querying for, with 3-4 negations this is not clear. (And if I parsed them correctly, 'NULL' can be replaced with $talent['talentTreeId'])